src/sdk/namespace_blob.js

```diff
@@ -215,7 +215,7 @@ class NamespaceBlob {
                         inspect(_.omit(params, 'object_md.ns')),
                         'callback res', inspect(res)
                     );
-                    return resolve(read_stream.pipe(count_stream));
+                    return resolve(read_stream);
```

Comment:
I think that this `resolve` might be redundant since the promise is resolved when the stream becomes readable. in any case, I don't think you should resolve in two different ways in different flows. you can maybe pipe once when creating the stream